src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.WriteHelpers.cs

```diff
@@ -64,10 +64,11 @@ internal Task SerializeAsync(Stream utf8Json,
             CancellationToken cancellationToken,
             object? rootValueBoxed = null)
         {
+            PooledByteBufferWriter writer = new PooledByteBufferWriter(Options.DefaultBufferSize, utf8Json);
             // Value chosen as 90% of the default buffer used in PooledByteBufferWriter.
             // This is a tradeoff between likelihood of needing to grow the array vs. utilizing most of the buffer
-            int flushThreshold = (int)(Options.DefaultBufferSize * JsonSerializer.FlushThreshold);
-            return SerializeAsync(new PooledByteBufferWriter(Options.DefaultBufferSize, utf8Json), rootValue, flushThreshold, cancellationToken, rootValueBoxed);
+            int flushThreshold = (int)(writer.Capacity * JsonSerializer.FlushThreshold);
```

Comment:
While looking at the Flush count in https://github.com/dotnet/runtime/pull/116947/files#diff-23d631f9621b20ea342bc3cd322c9bd843c8f6ee7fcdabec05fc7e57c956b4f8L142 I noticed that the Stream overload was doing a single write of 500k, this was because while adding the PipeWriter overload we ended up not using the Capacity for the flush threshold and so we ended up with 0 for flush threshold which means never flush. Updating to use `PooledByteBufferWriter.Capacity` like we used before, and updated the test to check for more than 1 write call.